src/test/controllerscriptenginelegacy_test.cpp

```diff
@@ -745,3 +745,43 @@ TEST_F(ControllerScriptEngineLegacyTest, screenWillSentRawDataIfConfigured) {
     ASSERT_ALL_EXPECTED_MSG();
 }
 #endif
+
+TEST_F(ControllerScriptEngineLegacyTest, convertCharsetUndefinedOnUnknownCharset) {
+    logMessagesExpected.append(std::tuple<QtMsgType, QRegularExpression>(
+            QtMsgType::QtWarningMsg, QStringLiteral("Unable to open encoder")));
```

Comment:
Using something like `try { } catch { }` is the JS code?

Reply:
Rather just seeing if instead of a QByteArray the QJSValue returns an error using `QJSValue::isError`. Haven't tried that yet, but I'm fairly certain it should work.

Reply:
Ok, so as it turns out, [`SETUP_LOG_CAPTURE`](https://github.com/mixxxdj/mixxx/blob/main/src/test/controllerscriptenginelegacy_test.cpp#L674C5-L674C22) was responsible. What hinted me was that [other earlier tests](https://github.com/mixxxdj/mixxx/blob/main/src/test/controllerscriptenginelegacy_test.cpp#L122-L126), also calling `logOrThrowError` weren't failing. Since there doesn't seem to be a `qUninstallMessageHandler`, I just moved the tests before the first `SETUP_LOG_CAPTURE`.

Tests are passing locally so I suppose this PR is ready for another review.